src/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv/Internal/LibuvOutputConsumer.cs

```diff
@@ -65,6 +65,11 @@ public class LibuvOutputConsumer
                         }
                     }
 
+                    if (_socket.IsClosed)
```

Comment:
Gross. The OS should timeout writes for completely unresponsive clients to begin with. Long term, the better solution is to enforce a minimum minimum data rate for responses.

This might require a way to cancel LibuvAwaitables, but it definitely not OK to immediately kill the socket and any ongoing writes just because the request processing loop completed.

You could consider doing that after some kind of timeout, but at that point you might as well just enforce a minimum rate.